File: src/main/java/com/intellij/plugin/applescript/lang/dictionary/xml/LegacyJdomParser.java
package com.intellij.plugin.applescript.lang.dictionary.xml;

import com.intellij.openapi.util.JDOMUtil;
import org.jdom.Document;
import org.jdom.JDOMException;

import java.io.File;
import java.io.IOException;

public final class LegacyJdomParser {
    private LegacyJdomParser() {
    }

    public static Document build(File file) throws JDOMException, IOException {
        return new Document(JDOMUtil.load(file.toPath()));
    }
}

File: src/main/kotlin/com/intellij/plugin/applescript/lang/dictionary/xml/LegacyJdomParser.kt
package com.intellij.plugin.applescript.lang.dictionary.xml

import com.intellij.openapi.util.JDOMUtil
import org.jdom.Document
import java.io.File

/** Reads a dictionary XML file into a JDOM [Document], the shape the SDEF ingest pipeline expects. */
internal object LegacyJdomParser {
    fun build(file: File): Document = Document(JDOMUtil.load(file.toPath()))
}
